eating_cookies/eating_cookies.py

```diff
@@ -3,14 +3,74 @@
 import sys
 
 # The cache parameter is here for if you want to implement
-# a solution that is more efficient than the naive 
+# a solution that is more efficient than the naive
 # recursive solution
-def eating_cookies(n, cache=None):
-  pass
+
+
+def eating_cookies(n, cache=[]):
+    ways = [0] * (n + 1)
+    start = [1, 1, 2]
+    for i in range(len(ways)):
+        if i <= 2:
+            ways[i] = start[i]
+        else:
+            ways[i] = ways[i-1] + ways[i-2] + ways[i-3]
+    return ways[n]
```

Comment:
Nice work using an iterative solution that builds the sequence up.  I personally preferred this method myself as well compared to recursion or anything else.